nominal/experimental/migration/migration_utils.py

```diff
@@ -217,33 +245,295 @@ def clone_workbook_template(
+                        v1=scout_layout_api.TabbedPanelV1(
+                            id=str(uuid.uuid4()),
+                            tabs=[],
+                        )
+                    )
+                )
+            )
+        )
+        new_workbook_content = scout_workbookcommon_api.WorkbookContent(channel_variables={}, charts={})
+    new_workbook_template: WorkbookTemplate = create_workbook_template_with_content_and_layout(
+        client=destination_client,
+        title=new_template_title or raw_source_template.metadata.title,
+        description=new_template_description or raw_source_template.metadata.description,
+        labels=new_template_labels or raw_source_template.metadata.labels,
+        properties=new_template_properties or raw_source_template.metadata.properties,
         layout=new_template_layout,
         content=new_workbook_content,
         commit_message="Cloned from template",
-        workspace_rid=workspace_rid,
+        workspace_rid=destination_client.get_workspace().rid,
+    )
+    logger.debug(
+        "New workbook template created %s from %s (rid: %s)",
+        new_workbook_template.title,
+        source_template.title,
+        source_template.rid,
+        extra=log_extras,
+    )
+    return new_workbook_template
+
+
+def copy_file_to_dataset(
+    source_file: DatasetFile,
+    destination_dataset: Dataset,
+) -> DatasetFile:
+    """Copy a dataset file from the source to the destination dataset.
+
+    Args:
+        source_file: The source DatasetFile to copy.
+        destination_dataset: The Dataset to create the copied file in.
+
+    Returns:
+        The dataset file in the new dataset.
+    """
+    log_extras = {"destination_client_workspace": destination_dataset._clients.workspace_rid}
+    logger.debug("Copying dataset file: %s", source_file.name, extra=log_extras)
+    source_api_file = source_file._get_latest_api()
+    if (
+        source_api_file.handle.s3 is not None
+        and source_file.timestamp_channel is not None
+        and source_file.timestamp_type is not None
+    ):
+        old_file_uri: str = source_file._clients.catalog.get_dataset_file_uri(
+            source_file._clients.auth_header, source_file.dataset_rid, source_file.id
+        ).uri
+
+        response = requests.get(old_file_uri, stream=True)
+        response.raise_for_status()
+
+        with closing(BytesIO()) as file_obj:
+            # TODO (Sean): Stream files so we don't have to load objects fully into memory
+            for chunk in response.iter_content(chunk_size=8192):
+                file_obj.write(chunk)
+            file_obj.seek(0)
+
+            file_name: str = source_api_file.handle.s3.key.split("/")[-1]
+            file_type: FileType = FileType.from_path(file_name)
+            file_stem: str = Path(file_name).stem
+
+            new_file: DatasetFile = destination_dataset.add_from_io(
+                dataset=file_obj,
```

Comment:
Why TODO? With `stream=True`, then `response.content` will be a file-like object so I think you can simply do `dataset=response.content`